Duplicati/Library/Main/Operation/RestoreHandler.cs

```diff
@@ -610,8 +610,8 @@ private static void ScanForExistingSourceBlocksFast(LocalRestoreDatabase databas
                                             if (sourcestream.Length > block.Offset)
                                             {
                                                 sourcestream.Position = block.Offset;
-                                                
-                                                var size = sourcestream.Read(blockbuffer, 0, blockbuffer.Length);
+
+                                                int size = Library.Utility.Utility.ForceStreamRead(sourcestream, blockbuffer, blockbuffer.Length);
                                                 if (size == block.Size)

```

Comment:
This condition feels odd to me.  Is `sourcestream` guaranteed to be `block.Size` bytes in length?

Reply:
As I understand it, it checks if the data we just read into the buffer has the same length as the block we are looking for. In theory, we could use matching data where we just read `block.size` bytes each time, but that produces a number of non-aligned blocks which is generally not supported as it increases the processing time dramatically.
As an example, if `block.size == 1` we would read a single byte and hash it each time, where now we read full blocks until EOF.